### DISCUSSION TITLE
pythoncom.CoGetInterfaceAndReleaseStream throws exception in the second time

### DISCUSSION THREAD

**Discussion by ahmad811** [mhammond/pywin32#2550] [General]
Note that issues in this repository are only for bugs or feature requests in the pywin32.

**If you need support or help using this package, please follow [these instructions](https://github.com/mhammond/pywin32/blob/master/README.md#support)** - support or help requests will be closed without comment.

For all bugs, please provide the following information.

* Expected behavior and actual behavior.
I want to call MyOle in different threads...
The second myOle.call1 will throw an exception: args:(-2147417827, 'The marshaled inter...n format.', None, None)

* Steps to reproduce the problem.
In the main.py:
`
class My_HTTPServer_RequestHandler(BaseHTTPRequestHandler):
    def do_POST(self):
        self.callFoo()

    def callFoo(self):
        myOle = MyOleWrapper()
        myOle.create()
        myOle.call1()
        # This will throw an exception
        myOle.call1()

def run():
    server_address = ('', 80)
    httpd = ThreadingHTTPServer(server_address,My_HTTPServer_RequestHandler)
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        pass
    httpd.server_close()
 
if __name__ == "__main__":
    run()
`

In MyOleWrapper.py
`
from win32com import client
import pythoncom
class MyOleWrapper():
    
    def create(self):
        pythoncom.CoInitialize()
        ole = client.Dispatch('what ever')
        oleid = pythoncom.CoMarshalInterThreadInterfaceInStream(pythoncom.IID_IDispatch, old)
        return oleid

    def call1(self, oleid):
        pythoncom.CoInitialize()
        x= client.Dispatch(pythoncom.CoGetInterfaceAndReleaseStream(oleid, pythoncom.IID_IDispatch))
        x = client.Dispatch(x)
        # do some stuff with x
        pythoncom.CoUninitialize()
`
* Version of Python and pywin32
python: 3.7.3
pywin32: 223

**Comment by mhammond** [comment]
This is COM throwing the error because it has strict rules around threading - ie, this is a support request, so you need to follow the instructions you quoted at the top of your issue.


> **If you need support or help using this package, please follow [these instructions](https://github.com/mhammond/pywin32/blob/master/README.md#support)** - support or help requests will be closed without comment.
